### PR TITLE
Docs: update document release steps

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Remove the references about archived `iceberg-docs` repo and point to README.

### CODE REVIEW

File: site/docs/how-to-release.md [hunk 1]
```diff
@@ -311,74 +311,7 @@ Create a PR in the `iceberg` repo to make revapi run on the new release. For an
...
-
-```shell
-git checkout latest
-git rebase main
-git push apache latest
-```
-
-#### Set latest version in iceberg-docs repo
-
-The last step is to update the `main` branch in `iceberg-docs` to set the latest version.
```
Comment (ajantha-bhat) [inline site/docs/how-to-release.md:373]: @bitsondatadev: I didn't see any info about this in README now. 
Also couldn't find these config.toml etc in the orphan docs and iceberg repo. Any idea how to update these now?

Comment (ajantha-bhat) [reply site/docs/how-to-release.md:373]: Never mind. I think we should do like this https://github.com/apache/iceberg/pull/9931

Comment (ajantha-bhat) [reply site/docs/how-to-release.md:373]: Nope. Nessie version is still not resolved in the latest site.
https://iceberg.apache.org/docs/latest/nessie/#spark-sql-extensions

Comment (bitsondatadev) [reply site/docs/how-to-release.md:373]: We need to add that file here next to the `icebergVersion` under `extra`: https://github.com/apache/iceberg/blob/main/site/mkdocs.yml

Comment (ajantha-bhat) [reply site/docs/how-to-release.md:373]: Done. Update. Please check and approve the PR if it is good enough.

File: site/docs/how-to-release.md [hunk 2]
```diff
@@ -311,74 +311,9 @@ Create a PR in the `iceberg` repo to make revapi run on the new release. For an
 ### Documentation Release
 
 Documentation needs to be updated as a part of an Iceberg release after a release candidate is passed.
-The commands described below assume you are in a directory containing a local clone of the `iceberg-docs`
-repository and `iceberg` repository. Adjust the commands accordingly if it is not the case. Note that all
-changes in `iceberg` need to happen against the `master` branch and changes in `iceberg-docs` need to happen
-against the `main` branch. 
+Please follow the steps from [README](../README.md) to update the site docs and javadocs.
```
Comment (nastra) [inline site/docs/how-to-release.md:314]: should this link to the section that mentions what needs to be done during a release?

Comment (ajantha-bhat) [reply site/docs/how-to-release.md:314]: I wanted to link to https://github.com/apache/iceberg/blob/main/site/README.md#release-process, but `release deploy` won't work. So, I linked the whole file. 

P.S: I just want to get rid of referencing the archived `iceberg-docs` repo. Once we have automation we can update this section or add more info about manual steps in a follow up.

Comment (bitsondatadev) [reply site/docs/how-to-release.md:314]: Ajantha, now that the release is out, could you and @Fokko and @nastra help me review my release script?

Three thinks I'd like to have fixed by the next release are:

1. Fix internal links that cross versioned doc and site boundaries.

1. Add release script.

1. Update release and contribution docs.

Comment (ajantha-bhat) [reply site/docs/how-to-release.md:314]: Always happy to validate. 

So you want us to keep this PR on hold until we have the script for release?

**Comment by ajantha-bhat** [comment]
Closing this PR as https://github.com/apache/iceberg/pull/10402 has handled the same.

**Pull request closed without merging**